driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -405,23 +402,6 @@ private void connectionClosed(final ConnectionPoolListener connectionPoolListene
         return removedReason;
     }
 
-    /**
-     * If there was a socket exception that wasn't some form of interrupted read, increment the generation count so that any connections
-     * created prior will be discarded.
-     *
-     * @param connection the connection that generated the exception
-     * @param t          the exception
-     */
-    private void incrementGenerationOnSocketException(final InternalConnection connection, final Throwable t) {
-        if (t instanceof MongoSocketException && !(t instanceof MongoSocketReadTimeoutException)) {
```

Comment:
So these type exceptions are now handled at the Default Server level?  I would have thought SDAM spec says that socket exceptions invalidate the pool.

Reply:
Yes, handled at DefaultServer.  It was actually always handled there.  And here.  And we were incrementing the generation twice.  We only noticed the double increment when we started comparing the connection generation with the pool generation and getting incorrect behavior as a result. The `connection.generation = pool.generation` condition in DefaultServer was never true in `DefaultServer` because we had already incremented the pool's generation by the time the condition was tested.  By removing it here, that test starts to do what we intended.

Reply:
Ack